### ISSUE TITLE
Incorrect case-sensitive schema name handling when dropping the view

### ISSUE THREAD

**Issue by staticlibs** [wiltondb/wiltondb#57]
The following script with uppercase schema `[SCH2]` fails to re-create the `[SCH2].v2` view after the drop:

```tsql
create database test1
go
use test1
go

create schema sch1
go
create view sch1.v1 as select 42
go
drop view sch1.v1
go
create view sch1.v1 as select 42
go

create schema [SCH2]
go
create view [SCH2].v2 as select 42
go
drop view [SCH2].v2
go
create view [SCH2].v2 as select 42
go
```
```
Msg 2627, Level 14, State 1, Line 21
duplicate key value violates unique constraint "babelfish_view_def_pkey"
```

The same script works correctly when `[sch2]` schema is created in lower-case:

```tsql
create database test1
go
use test1
go

create schema sch1
go
create view sch1.v1 as select 42
go
drop view sch1.v1
go
create view sch1.v1 as select 42
go

create schema [sch2]
go
create view [SCH2].v2 as select 42
go
drop view [SCH2].v2
go
create view [SCH2].v2 as select 42
go
```

**Comment by psrinivasa2** [comment]
Hi Alex,
My scenario is different,

Working Scenario:

1) Login into SSMS via OS user 
2) Execute the database schema
3) Try test the drop view and recreate, works fine without issue

Not Working Scenario:

1) Login into SSMS via OS user 
2) Execute the database schema
3) Connect via application
4) Common script file which I mentioned will be executed as part of application code (still application is running under OS user only)
5) Now error occurs (i.e drop view not removed from sys.babelfish_view_def)
6) then anytime I connect SSMS, the error remains

We need to see, why the issues occurs when run via application

**Comment by staticlibs** [comment]
@psrinivasa2 

Does the problem occur if you create the problematic schema in lower-case in DDL script?

**Comment by psrinivasa2** [comment]
Yes Alex, I tried create in lower case and still the same scenario when running in application

**Comment by staticlibs** [comment]
@psrinivasa2 

When the problem happens, please check the schema name case in `sys.babelfish_namespace_ext` table (`orig_name` column) and in `sys.babelfish_view_def` (`shema_name` column) table. The name must be lower-case there in both cases. If the schema name is already lower-case everywhere in DB - then your problem indeed can be different from the one listed in this issue (I've created that minimal example by simplifying your original script from email).

The problem is tricky to reproduce, because it is tied to cache initialization - note that `sch1` schema (that is not linked to `SCH2` in any way) code block in example above is mandatory - problem won't happen without `sch1` block run before `SCH2` one. I assume the problem happens in app, but not in SSMS for you because of the previous statements executed in the same DB session.

**Comment by psrinivasa2** [comment]
Hi Alex,
I have several schema's and I tried by lower casing only the one schema which was problematic. Let me give a try on lower casing all the schema name and check

**Comment by psrinivasa2** [comment]
Hi Alex, @staticlibs ,

You are correct , after making all schema's too lower case, it works fine. So casing is the issue

**Comment by psrinivasa2** [comment]
Helpful if you add the fix on this to maintain the meaningful casing for schema name's

**Comment by staticlibs** [comment]
@psrinivasa2

The root of the problem appeared to be not in cache, but in incorrect collation used in `babelfish_view_def` queries, so the index search was basically returning randomly incorrect result. I will include the fix into the upcoming release, the workaround with lower-case schema (still effective) won't be necessary with this fix. Assuming no major problems discovered in the following days - going to prepare the updated release (with all pending changes) during the next week.

**Comment by staticlibs** [comment]
The fix is included in [update 12.16.1](https://github.com/wiltondb/wiltondb/releases/tag/3-lts-12-16-1).

**Comment by psrinivasa2** [comment]
This can be closed
